execution-core/src/transfer/phoenix.rs

```diff
@@ -273,6 +273,13 @@ impl Transaction {
         Self { payload, proof }
     }
 
+    /// Replaces the inner `proof` bytes for a given `proof`.
+    /// Note: This method is likely to invalidate the transaction and should
+    /// only be used with care.
+    pub fn replace_proof(&mut self, proof: Vec<u8>) {
```

Comment:
Why don't we flag the method as unsafe?

Reply:
It's not really unsafe in rust's sense. I added more substance to the comment though!